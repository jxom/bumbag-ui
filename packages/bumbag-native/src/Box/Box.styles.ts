import { styled } from '../styled';
import { theme } from '../utils/theme';

export const StyledBox = styled.View`
  ${theme('native.Box', 'styles.base')};
`;

export const StyledBoxSafe = styled.SafeAreaView`
  ${theme('native.Box.Safe', 'styles.base')};
`;

export const StyledBoxScroll = styled.ScrollView`
  ${theme('native.Box.Scroll', 'styles.base')};
`;

export const StyledBoxKeyboardAvoiding = styled.KeyboardAvoidingView`
  ${theme('native.Box.KeyboardAvoiding', 'styles.base')};
`;

export const StyledBoxTouchable = styled.TouchableOpacity`
  ${theme('native.Box.Touchable', 'styles.base')};
`;

export const StyledBoxTouchableWithoutFeedback = styled.TouchableWithoutFeedback`
  ${theme('native.Box.TouchableWithoutFeedback', 'styles.base')};
`;
