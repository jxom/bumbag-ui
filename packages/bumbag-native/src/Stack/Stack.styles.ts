import { styled } from '../styled';
import { theme } from '../utils/theme';

export const StyledStack = styled.View`
  ${(props: any) =>
    props.orientation === 'horizontal'
      ? `
          flex-direction: row;
        `
      : ''}

  ${theme('native.Stack', 'styles.base')};
`;
