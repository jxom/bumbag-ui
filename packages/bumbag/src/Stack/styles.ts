import { css, cssClass } from '../styled';
import { space, theme } from '../utils';
import { getFlexAlignmentAttributes } from '../Flex/styles';

const verticalBreakpoints = {
  tablet: 'mobile',
  desktop: 'tablet',
  widescreen: 'desktop',
  fullHD: 'widescreen',
};

export const Stack = (styleProps) => cssClass`
  ${styleProps.orientation === 'vertical' && getVerticalAttributes(styleProps)}
  ${styleProps.orientation === 'horizontal' && getHorizontalAttributes(styleProps)}

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;

const getVerticalAttributes = (styleProps) => css`
  && > *:not(:last-child) {
    margin-bottom: ${space(styleProps.spacing)(styleProps)}rem;

    & {
      ${theme(styleProps.themeKey, `styles.child.vertical`)(styleProps)};
    }
  }

  & {
    ${theme(styleProps.themeKey, `styles.vertical`)(styleProps)};
  }
`;

const getHorizontalAttributes = (styleProps) => {
  let breakpoint = theme('breakpoints', verticalBreakpoints[styleProps.verticalBelow])(styleProps);
  breakpoint = breakpoint ? `${breakpoint}px` : styleProps.verticalBelow;
  return css`
    @media screen and (min-width: ${breakpoint}) {
      display: flex;

      ${!styleProps.alignX &&
      !styleProps.alignY &&
      css`
        & > * {
          flex: 1;
        }
      `}

      ${getFlexAlignmentAttributes(styleProps)}

      & > *:not(:last-child) {
        margin-right: ${space(styleProps.spacing)(styleProps)}rem;

        & {
          ${theme(styleProps.themeKey, `styles.child.horizontal`)(styleProps)};
        }
      }

      & {
        ${theme(styleProps.themeKey, `styles.horizontal`)(styleProps)};
      }
    }

    @media screen and (max-width: ${breakpoint}) {
      ${getVerticalAttributes(styleProps)};
    }
  `;
};
