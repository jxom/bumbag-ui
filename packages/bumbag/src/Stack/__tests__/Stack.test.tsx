import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Box } from '../../Box';
import { Stack } from '../Stack';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Stack>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <Stack color="primary">
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for a horizontal set', () => {
    const { container } = render(
      <Stack orientation="horizontal">
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for a horizontal set with breakpoint', () => {
    const { container } = render(
      <Stack orientation="horizontal" verticalBelow="widescreen">
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  ['left', 'right', 'center'].forEach((alignX) => {
    it(`should render correctly for an alignX of ${alignX}`, () => {
      const { container } = render(
        // @ts-ignore
        <Stack alignX={alignX}>
          <Box>hello</Box>
          <Box>world</Box>
        </Stack>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  ['top', 'bottom', 'center'].forEach((alignY) => {
    it(`should render correctly for an alignY of ${alignY}`, () => {
      const { container } = render(
        // @ts-ignore
        <Stack alignY={alignY}>
          <Box>hello</Box>
          <Box>world</Box>
        </Stack>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Stack use="div">
          <Box>hello</Box>
          <Box>world</Box>
        </Stack>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Stack props', () => {
      const { result } = renderHook(() => Stack.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Stack>
          {(StackProps) => (
            <div {...StackProps}>
              <Box>hello</Box>
              <Box>world</Box>
            </div>
          )}
        </Stack>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('Stack.base should render correctly', () => {
    const { container } = render(
      <Stack overrides={{ Stack: { styles: { base: { backgroundColor: 'red' } } } }}>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.child.base should render correctly', () => {
    const { container } = render(
      <Stack overrides={{ Stack: { styles: { child: { base: { backgroundColor: 'red' } } } } }}>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.vertical should render correctly', () => {
    const { container } = render(
      <Stack overrides={{ Stack: { styles: { vertical: { backgroundColor: 'red' } } } }}>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.horizontal should render correctly', () => {
    const { container } = render(
      <Stack orientation="horizontal" overrides={{ Stack: { styles: { horizontal: { backgroundColor: 'red' } } } }}>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.child.vertical should render correctly', () => {
    const { container } = render(
      <Stack overrides={{ Stack: { styles: { child: { vertical: { backgroundColor: 'red' } } } } }}>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.child.horizontal should render correctly', () => {
    const { container } = render(
      <Stack
        orientation="horizontal"
        overrides={{ Stack: { styles: { child: { horizontal: { backgroundColor: 'red' } } } } }}
      >
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Stack.styles.base should render correctly', () => {
    const { container } = render(
      <Stack>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>,
      {
        theme: { Stack: { styles: { base: { backgroundColor: 'red' } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.styles.child.base should render correctly', () => {
    const { container } = render(
      <Stack>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>,
      {
        theme: { Stack: { styles: { child: { base: { backgroundColor: 'red' } } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.styles.vertical should render correctly', () => {
    const { container } = render(
      <Stack>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>,
      {
        theme: { Stack: { styles: { vertical: { backgroundColor: 'red' } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.styles.horizontal should render correctly', () => {
    const { container } = render(
      <Stack>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>,
      {
        theme: { Stack: { styles: { horizontal: { backgroundColor: 'red' } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.styles.child.vertical should render correctly', () => {
    const { container } = render(
      <Stack>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>,
      {
        theme: { Stack: { styles: { child: { vertical: { backgroundColor: 'red' } } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Stack.styles.child.horizontal should render correctly', () => {
    const { container } = render(
      <Stack>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>,
      {
        theme: { Stack: { styles: { child: { horizontal: { backgroundColor: 'red' } } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(
      <Stack>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>,
      {
        theme: { Stack: { defaultProps: { className: 'test' } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for orientation', () => {
    const { container } = render(
      <Stack>
        <Box>hello</Box>
        <Box>world</Box>
      </Stack>,
      {
        theme: { Stack: { defaultProps: { orientation: 'horizontal' } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
