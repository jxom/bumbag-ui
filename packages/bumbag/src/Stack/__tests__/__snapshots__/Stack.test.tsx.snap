// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`composition hook should return with Stack props 1`] = `
Object {
  "className": "css-ucri7v css-1pwrqjj css-1yuzjtp fp-Box fp-Stack",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

<div
  class="emotion-4 emotion-0 emotion-1 test fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`defaultProps should render correctly for orientation 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

@media screen and (min-width:768px) {
  .emotion-4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }

  .emotion-4 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-4 > *:not(:last-child) {
    margin-right: 2rem;
  }
}

@media screen and (max-width:768px) {
  .emotion-4.emotion-4 > *:not(:last-child) {
    margin-bottom: 2rem;
  }
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`overrides Stack.base should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-4 {
  background-color: #ff0000;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`overrides Stack.child.base should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`overrides Stack.child.horizontal should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

@media screen and (min-width:768px) {
  .emotion-4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }

  .emotion-4 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-4 > *:not(:last-child) {
    margin-right: 2rem;
  }

  .emotion-4 > *:not(:last-child) {
    background-color: #ff0000;
  }
}

@media screen and (max-width:768px) {
  .emotion-4.emotion-4 > *:not(:last-child) {
    margin-bottom: 2rem;
  }
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`overrides Stack.child.vertical should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  background-color: #ff0000;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`overrides Stack.horizontal should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

@media screen and (min-width:768px) {
  .emotion-4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }

  .emotion-4 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-4 > *:not(:last-child) {
    margin-right: 2rem;
  }

  .emotion-4 {
    background-color: #ff0000;
  }
}

@media screen and (max-width:768px) {
  .emotion-4.emotion-4 > *:not(:last-child) {
    margin-bottom: 2rem;
  }
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`overrides Stack.vertical should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-4 {
  background-color: #ff0000;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly for a horizontal set 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

@media screen and (min-width:768px) {
  .emotion-4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }

  .emotion-4 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-4 > *:not(:last-child) {
    margin-right: 2rem;
  }
}

@media screen and (max-width:768px) {
  .emotion-4.emotion-4 > *:not(:last-child) {
    margin-bottom: 2rem;
  }
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly for a horizontal set with breakpoint 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

@media screen and (min-width:1024px) {
  .emotion-4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }

  .emotion-4 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-4 > *:not(:last-child) {
    margin-right: 2rem;
  }
}

@media screen and (max-width:1024px) {
  .emotion-4.emotion-4 > *:not(:last-child) {
    margin-bottom: 2rem;
  }
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly for an alignX of center 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="emotion-4 emotion-0 emotion-6 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly for an alignX of left 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  class="emotion-4 emotion-0 emotion-6 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly for an alignX of right 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

<div
  class="emotion-4 emotion-0 emotion-6 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly for an alignY of bottom 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  class="emotion-4 emotion-0 emotion-6 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly for an alignY of center 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="emotion-4 emotion-0 emotion-6 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly for an alignY of top 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  class="emotion-4 emotion-0 emotion-6 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-6 {
  color: #574feb !important;
}

<div
  class="emotion-4 emotion-0 emotion-6 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`theming Stack.styles.base should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-4 {
  background-color: #ff0000;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`theming Stack.styles.child.base should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`theming Stack.styles.child.horizontal should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`theming Stack.styles.child.vertical should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  background-color: #ff0000;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`theming Stack.styles.horizontal should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;

exports[`theming Stack.styles.vertical should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4.emotion-4 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.emotion-4 {
  background-color: #ff0000;
}

<div
  class="emotion-4 emotion-0 emotion-1 fp-Box fp-Stack"
>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    hello
  </div>
  <div
    class="emotion-0 emotion-1 fp-Box"
  >
    world
  </div>
</div>
`;
